src/App.tsx

```diff
@@ -14,11 +15,98 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  visibleStart: boolean,
+  resetGoods: string[],
+  goods: string[],
+};
+
+class App extends React.Component<{}, State> {
+  state = {
+    visibleStart: false,
+    resetGoods: [...goodsFromServer],
+    goods: [...goodsFromServer],
+  };
+
+  render() {
+    const { visibleStart, goods } = this.state;
+
+    return (
+      <div className="good">
+        <div className="container">
+          <h1 className="good__title title is-3">List of goods</h1>
+          <button
+            type="button"
+            onClick={() => (this.setState({ visibleStart: true }))}
+            className={classNames(
+              'button is-warning good__start-button is-medium',
+              {
+                'is-hidden': visibleStart === true,
+              },
+            )}
+          >
+            Start
+          </button>
+          <div className="good__list-block">
+            {visibleStart && (
+              <>
+                <ul className="good__list">
+                  {goods.map(good => (
+                    <li
+                      key={good}
+                      className="good__item subtitle is-3"
+                    >
+                      {good}
+                    </li>
+                  ))}
+                </ul>
+                <button
+                  type="button"
+                  onClick={() => this.setState(state => ({
```

Comment:
It'll be better if you write seperate method - render(), for example, - and set this method in onClick.